packages/firestore/src/util/input_validation.ts

```diff
@@ -292,6 +291,29 @@ export function validateNamedOptionalPropertyEquals<T>(
   }
 }
 
+/**
+ * Validates that the provided argument is a valid enum.
+ *
+ * @param functionName Function making the validation call.
+ * @param enums Array containing all possible values for the enum.
+ * @param position Position of the argument in `functionName`.
+ * @param argument Arugment to validate.
+ */
+export function validateStringEnum<T>(
+  functionName: string,
+  enums: string[],
+  position: number,
+  argument: unknown
+): void {
+  if (!enums.some(element => element === argument)) {
+    throw new FirestoreError(
+      Code.INVALID_ARGUMENT,
+      `Invalid value ${argument} provided to function ${functionName}() for its` +
```

Comment:
I'd wrap ${argument} in quotes to clearly delineate it from the rest of the sentence and make the error clearer e.g. if the user accidentally passes an empty string they should see `Invalid value "" provided ...` instead of `Invalid value  provided ...`.

Or to make sure the error is relatively readable for non-strings as well, you could use the `valueDescription()` helper method that we use elsewhere (which wraps strings in quotes as well as massages other types into readable descriptions).